src/modules/CurrencyModule.cpp

```diff
@@ -1,46 +1,350 @@
+			return;
+		}
+
+		if (args.size() == 1) {
+			giveawayChannelId = message.channelId();
+			client.createMessage(message.channelId(), "**Giveaway announcement channel successfully changed to current channel.**");
+		}
+
+		});
+
+	RegisterCommand(Commands::CURRENCY_SET_NAME, "setcurrenname", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel) {
+		
+		QStringList args = message.content().split(' ');
+		GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+		QString prefix = setting->prefix;
+
+		if (args.first() != prefix + "setcurrenname")
+			return;
+
+		if (args.size() > 2) {
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+
+		if (args.size() == 2) {
+			currencyName = args.at(1);
+			client.createMessage(message.channelId(), "**Currency Name set to** " + currencyName);
+		}
+
+		});
+
+	RegisterCommand(Commands::CURRENCY_SET_SYMBOL, "setcurrensymb", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel) {
+
+		QStringList args = message.content().split(' ');
+		GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+		QString prefix = setting->prefix;
+
+		if (args.first() != prefix + "setcurrensymb")
+			return;
+
+		if (args.size() > 2) {
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+
+		if (args.size() == 2) {
+			currencySymbol = args.at(1);
+			client.createMessage(message.channelId(), "**Currency Symbol set to** " + currencySymbol);
+		}
+
+		});
 }
 
-void CurrencyModule::StatusCommand(QString& result, snowflake_t guild, snowflake_t user)
-{
-	result += "$ " + QString::number(m_settings[user].currency) + "\n";
-	result += "\n";
+void CurrencyModule::OnMessage(Discord::Client& client, const Discord::Message& message) {
+
+	if (!isRandomGiveAwayDone) {
+		static int pingCount = 0;
+		if (pingCount == 0) {
+			bool outcome = randp(random_engine);
+			if (outcome) {
+				client.createMessage(giveawayChannelId, "@everyone **FREEBIE** available now! Go `!claim` some juicy coins!");
+				allowGiveaway = true;
+				pingCount++;
+			}
+		}
+	}
+
+	if (selfGambleData.gamble && !message.author().bot() && message.channelId() == selfGambleData.channelId) {
+		if (message.author().id() == selfGambleData.userId) {
+
+			//Check if the message is a number
+			QRegExp re("\\d*");
+			if (!re.exactMatch(message.content()))
+				return;
+
+			GuildSetting* setting = &GuildSettings::GetGuildSetting(selfGambleData.guildId);
+			if (message.content().toInt() > 5 || message.content().toInt() < 0) {
+
+				client.createMessage(message.channelId(), "**Your guess is out of range!** \nTry a number between 0 and 5 (inclusive)");
+				return;
+
+			}
+			if (message.content().toInt() == selfGambleData.randNum) {
+				m_settings[setting->id].currency += 50;
```

Comment:
I think I would like all of these magic numbers as configurable variables. And by "all" I mean minimum roll (0), maximum roll (5), winner reward (50) and loser loss (10).

And since the person is gambling, can't we make it so that they gamble a given amount of currency for a chance to double it? I think that could be exciting!

Reply:
Sure, I will get rid of all the magic numbers... Do you want me to make them changeable using commands as well?

Reply:
And yeah the doubling thing would be quite exciting, I will work on it now!

Reply:
Yes if it is not too much to ask, please make them settable via a command!